### PR TITLE
Fixes #326

### LINKED ISSUE #326: Buildout 3.6.0 always offers add itself and setuptools into the versions section.
Buildout 3.6.0 always offers add itself and setuptools into the versions section if the show-picked-versions is true.
For example:
```
# buildout.cfg
[buildout]
show-picked-versions = true
versions = versions

[versions]
setuptools = 33.1.1
zc.buildout = 2.6.0
```
Console:
```
$ ./bin/buildout
...
Versions had to be automatically picked.
The following part definition lists the versions picked:
[versions]
setuptools = 33.1.1
zc.buildout = 2.6.0
```

### PR DESCRIPTION
**Pull request by jimfulton**
Fixed a bug introduced in 2.6.0:
zc.buildout and its dependeoncies were reported as picked even when
their versions were fixed in a ``versions`` section.  Worse, when the
``update-versions-file`` option was used, the ``versions`` section was
updated needlessly on every run.

### CODE REVIEW

File: src/zc/buildout/tests.py [hunk 1]
```diff
@@ -2408,12 +2408,12 @@ def wont_downgrade_due_to_prefer_final():
     ... parts =
     ... ''')
 
-    >>> [v] = [l.split('= >=', 1)[1].strip()
+    >>> [v] = [str(l.split('= >=', 1)[1].strip())
```
Comment (jamadden) [inline src/zc/buildout/tests.py]: I'm curious why the `str` was needed in these two places?

File: src/zc/buildout/tests.py [hunk 2]
```diff
@@ -3062,6 +3062,39 @@ def test_abi_tag_eggs():
...
+    >>> _ = system(join('bin', 'buildout'))
+    >>> with open('versions.cfg') as f:
+    ...     versions = f.read()
+    >>> _ = system(join('bin', 'buildout'))
+    >>> _ = system(join('bin', 'buildout'))
+    >>> _ = system(join('bin', 'buildout'))
+    >>> with open('versions.cfg') as f:
+    ...     versions == f.read()
+    True
+    >>> cat('versions.cfg') # doctest: +ELLIPSIS
```
Comment (jamadden) [inline src/zc/buildout/tests.py]: I assume there's no risk of ELLIPSIS hiding extra lines getting appended to the file.

File: src/zc/buildout/tests.py (hunk 1 above)
Comment (jimfulton) [inline src/zc/buildout/tests.py]: Because I got a spurious failure when one was unicode and the other a str in Python 2,

File: src/zc/buildout/tests.py (hunk 2 above)
Comment (jamadden) [inline src/zc/buildout/tests.py]: Never mind, that doesn't even make sense given the above equality test.

Comment (jimfulton) [inline src/zc/buildout/tests.py]: The check above makes sure that versions.cfg didn't change after the first run.  The check on the 3090 sanity checks that something was added on the first run.  I'll rearrange this to make it clearer.

File: src/zc/buildout/tests.py (hunk 1 above)
Comment (jamadden) [inline src/zc/buildout/tests.py]: Hmm. Well that seems odd. I don't blame you for fixing the test here though.

Comment (jimfulton) [inline src/zc/buildout/tests.py]: It's really odd, now that you mention it. :/

So odd I now can't reproduce it. WTF? reverted.

File: src/zc/buildout/tests.py (hunk 2 above)
Comment (jamadden) [inline src/zc/buildout/tests.py]: Thanks, I do find that much clearer.

**Comment by jimfulton** [comment]
OK, I made the logic for controlling "picking" more explicit.

Riding ``dest`` was mostly OK mut could have some edge-case failures in offline mode, where dest is also None.

Buildout logic has evolved over the years, often under duress, and entropy has set in.

Thanks for fighting entropy. :)

**Comment by jamadden** [comment]
I figured this was a duress situation so I was trying not to be too picky :) 

Thanks for that change, that makes it a lot more clear to me.
